src/main/java/com/uid2/admin/vertx/service/ServiceLinkService.java

```diff
@@ -107,11 +109,35 @@ private void handleServiceLinkAdd(RoutingContext rc) {
                 return;
             }
 
+            Set<Role> serviceRoles = serviceProvider.getService(serviceId).getRoles();
+            final Set<Role> roles;
+            if (rolesJson == null || rolesJson.isEmpty()) {
+                // if no roles provided and service only allows for one role, populate automatically
+                if (serviceRoles.size() == 1) {
```

Comment:
Why is the case of 1 service role treated specially? What's wrong with always defaulting to service roles?

Reply:
If a service allows for multiple roles, we want participants to specify which role they have for the service link as they may only have access to a subset of it. For example, if the snowflake service allows for MAPPER and SHARER roles, the user may only have access to the role MAPPER. So, we want them to have to specify which role they have rather than allowing them to default to all service roles.